webhooks-extension/pkg/endpoints/webhook.go

```diff
@@ -17,32 +17,287 @@ import (
+		webhook.Pipeline+"-template",
+		webhook.GitRepositoryURL,
+		"push",
+		webhook.AccessTokenRef,
+		hookParams)
+
+	pullRequestTrigger := r.newTrigger(webhook.Name+"-"+webhook.Namespace+"-pullrequest-event",
+		webhook.Pipeline+"-pullrequest-binding",
+		webhook.Pipeline+"-template",
+		webhook.GitRepositoryURL,
+		"pull_request",
+		webhook.AccessTokenRef,
+		hookParams)
+
+	monitorTrigger := r.newTrigger(monitorTriggerName,
+		webhook.PullTask+"-binding",
+		webhook.PullTask+"-template",
+		webhook.GitRepositoryURL,
+		"pull_request",
+		webhook.AccessTokenRef,
+		monitorParams)
+
+	triggers := []v1alpha1.EventListenerTrigger{pushTrigger, pullRequestTrigger, monitorTrigger}
+
+	eventListener := v1alpha1.EventListener{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      eventListenerName,
+			Namespace: namespace,
+		},
+		Spec: v1alpha1.EventListenerSpec{
+			ServiceAccountName: "tekton-webhooks-extension-eventlistener",
+			Triggers:           triggers,
+		},
+	}
+
+	return r.TriggersClient.TektonV1alpha1().EventListeners(namespace).Create(&eventListener)
+}
+
+/*
+	Update of the eventlistener, called when adding additional webhooks as we
+	run with a single eventlistener.
+*/
+func (r Resource) updateEventListener(eventListener *v1alpha1.EventListener, webhook webhook, monitorTriggerName string) (*v1alpha1.EventListener, error) {
+	hookParams, monitorParams := r.getParams(webhook)
+	newPushTrigger := r.newTrigger(webhook.Name+"-"+webhook.Namespace+"-push-event",
+		webhook.Pipeline+"-push-binding",
+		webhook.Pipeline+"-template",
+		webhook.GitRepositoryURL,
+		"push",
+		webhook.AccessTokenRef,
+		hookParams)
+
+	newPullRequestTrigger := r.newTrigger(webhook.Name+"-"+webhook.Namespace+"-pullrequest-event",
+		webhook.Pipeline+"-pullrequest-binding",
+		webhook.Pipeline+"-template",
+		webhook.GitRepositoryURL,
+		"pull_request",
+		webhook.AccessTokenRef,
+		hookParams)
+
+	eventListener.Spec.Triggers = append(eventListener.Spec.Triggers, newPushTrigger)
+	eventListener.Spec.Triggers = append(eventListener.Spec.Triggers, newPullRequestTrigger)
+
+	existingMonitorFound := false
+	for _, trigger := range eventListener.Spec.Triggers {
+		if trigger.Name == monitorTriggerName {
+			existingMonitorFound = true
+			break
+		}
+	}
+	if !existingMonitorFound {
+		newMonitor := r.newTrigger(monitorTriggerName,
+			webhook.PullTask+"-binding",
+			webhook.PullTask+"-template",
+			webhook.GitRepositoryURL,
+			"pull_request",
+			webhook.AccessTokenRef,
+			monitorParams)
+
+		eventListener.Spec.Triggers = append(eventListener.Spec.Triggers, newMonitor)
+	}
+
+	return r.TriggersClient.TektonV1alpha1().EventListeners(eventListener.GetNamespace()).Update(eventListener)
+}
+
+func (r Resource) newTrigger(name, bindingName, templateName, repoURL, event, secretName string, params []pipelinesv1alpha1.Param) v1alpha1.EventListenerTrigger {
+	return v1alpha1.EventListenerTrigger{
+		Name: name,
+		Binding: v1alpha1.EventListenerBinding{
```

Comment:
The EventListenerTrigger has been changed.  The Binding is now Bindings and the type has been changed, too.

Reply:
Yes we are aware - currently this code prereqs 0.1 of triggers not master.  A decision is pending on whether we will move up given the tight release schedule of triggers 0.2 and when we might want to do a 0.3.